yarn.config.cjs

```diff
@@ -1,32 +1,328 @@
+          'NODE_OPTIONS=--experimental-vm-modules jest --verbose',
+        );
+
+        // All non-root packages must have the same "test:watch" script.
+        expectWorkspaceField(
+          workspace,
+          'scripts.test:watch',
+          'NODE_OPTIONS=--experimental-vm-modules jest --watch',
+        );
+      }
+
+      if (isChildWorkspace) {
+        // The list of files included in all non-root packages must only include
+        // files generated during the build process.
+        expectWorkspaceArrayField(workspace, 'files', 'dist/');
+      } else {
+        // The root package must specify an empty set of published files. (This
+        // is required in order to be able to import anything in
+        // development-only scripts, as otherwise the
+        // `node/no-unpublished-require` ESLint rule will disallow it.)
+        expectWorkspaceField(workspace, 'files', []);
+      }
+
+      // If one workspace package lists another workspace package within
+      // `dependencies` or `devDependencies`, the version used within the
+      // dependency range must match the current version of the dependency.
+      expectUpToDateWorkspaceDependenciesAndDevDependencies(Yarn, workspace);
+
+      // If one workspace package lists another workspace package within
+      // `peerDependencies`, the dependency range must satisfy the current
+      // version of that package.
+      expectUpToDateWorkspacePeerDependencies(Yarn, workspace);
+
+      // No dependency may be listed under both `dependencies` and
+      // `devDependencies`.
+      expectDependenciesNotInBothProdAndDev(
+        workspace,
+        dependenciesByIdentAndType,
+      );
+
+      // If one workspace package (A) lists another workspace package (B) in its
+      // `dependencies`, and B is a critical dependency package, then we need to ensure
+      // that B is also listed in A's `peerDependencies` and that the version
+      // range satisfies the current version of B.
+      const criticalDependencyPatterns = [
+        // Examples of patterns you might want to use:
+        // /-controller$/u, // for projects that use controllers
+        // /-plugin$/u,     // for projects that use plugins
+        // /-adapter$/u,    // for projects that use adapters
+        // Add any patterns specific to your project architecture
+      ];
+
+      expectCriticalDependenciesListedAsPeerDependencies(
+        Yarn,
+        workspace,
+        dependenciesByIdentAndType,
+        criticalDependencyPatterns,
+      );
+
+      // The root workspace (and only the root workspace) must specify the Yarn
+      // version required for development.
+      if (isChildWorkspace) {
+        workspace.unset('packageManager');
+      } else {
+        expectWorkspaceField(workspace, 'packageManager', 'yarn@4.2.2');
+      }
+
+      // All packages must specify a minimum Node.js version of 18.18.
+      expectWorkspaceField(workspace, 'engines.node', '^18.18 || >=20');
```

Comment:
@garrettbear - How do we stay up to date with yarn, node and other dependencies while being compatible with our consumers and making sure we align with Wallet Framework and other repos and MetaMask engineering teams?